src/fieldValidation.ts

```diff
@@ -1,21 +1,45 @@
+import Ajv from 'ajv';
+import addFormats from 'ajv-formats';
 import { phone } from 'phone';
 import { BaseFieldDataType } from './types';
 
+const ajv = new Ajv();
+addFormats(ajv, ['email', 'uri']);
+
+function validateEmail(email: string): boolean {
+	const emailSchema = {
+		type: 'string',
+		format: 'email',
+	};
+
+	return ajv.validate(emailSchema, email);
+}
+
+function validateUrl(url: string): boolean {
+	const uriSchema = {
+		type: 'string',
+		format: 'uri',
+	};
+	return ajv.validate(uriSchema, url);
+}
+
+function validatePhoneNumber(phoneNumber: string): boolean {
+	return phone(phoneNumber).isValid;
+}
+
 export const fieldValueIsValid = (
 	fieldValue: string,
 	dataType: BaseFieldDataType,
-) => {
+): boolean => {
 	switch (dataType) {
 		case BaseFieldDataType.NUMBER:
 			return /^[0-9]*$/.test(fieldValue);
 		case BaseFieldDataType.PHONE_NUMBER:
-			return phone(fieldValue).isValid;
+			return validatePhoneNumber(fieldValue);
```

Comment:
Super small change: these functions return a boolean!  In that case it's handy to name them as an `isFoo` statement (e.g. `isValidPhoneNumber`).  This signals super clearly that the return value is going to be a boolean, and also reads more nicely in things like if statements

(e.g. `if (isValidPhoneNumber) then ...`

Reply:
the sand lives! I think this is a reasonable choice, refactoring now

Reply:
_First we taught the humans how to ergonomically conform to humanity, then they taught us. Gradually we stopped evaluating their advice through our own experience and began trusting it, implicitly. The changes began slowly at first, so slowly no one noticed, the arc of the tree blown gently in a steady wind growing leeward. But soon enough a visitor to that land knew it as The Place of Glass-Eyed Peoples. And later, simply The Place of the Glass Eyes._